Frameworks/Starboard/EbrFile.cpp

```diff
@@ -0,0 +1,273 @@
+        iter = s_fileMap.find(s_maxFileId);
+    } while (iter != s_fileMap.end());
+
+    s_fileMap.emplace(s_maxFileId, std::move(file));
+
+    return s_maxFileId;
+}
+
+void EbrFile::RemoveFile(int fid) {
+    std::lock_guard<std::mutex> guard(s_fileMapLock);
+    s_fileMap.erase(fid);
+}
+
+//  IO funcs
+
+int EbrOpen(const char* file, int mode, int share) {
+    return EbrOpenWithPermission(file, mode, share, 0);
+}
+
+int EbrOpenWithPermission(const char* file, int mode, int share, int pmode) {
+
+    std::shared_ptr<EbrFile> fileToAdd;
+
+    // Special random number device. Just a stub.
+    fileToAdd = EbrDevRandomFile::CreateInstance(file, mode, share, pmode);
+    if (fileToAdd) {
+        return EbrFile::AddFile(std::move(fileToAdd));
+    } 
+    
+    // Special file type for cached storage files.
+    fileToAdd = EbrStorageFile::CreateInstance(file, mode, share, pmode);
+    if (fileToAdd) {
+        return EbrFile::AddFile(std::move(fileToAdd));
+    }
+
+    // No more special types. Assume its a real file.
+    fileToAdd = EbrIOFile::CreateInstance(file, mode, share, pmode);
+    return (fileToAdd) ? EbrFile::AddFile(std::move(fileToAdd)) : -1;
+}
+
+int EbrClose(int fd) {
+    EbrFile::RemoveFile(fd);
+    return 0;
+}
+
+int EbrFstat(int fd, struct stat* ret) {
+    return EbrFile::GetFile(fd)->Stat(ret);
+}
+
+int EbrFstat64i32(int fd, struct _stat64i32* ret) {
+    return EbrFile::GetFile(fd)->Stat64i32(ret);
+}
+
+intptr_t EbrGetOSFHandle(int fd) {
+    return EbrFile::GetFile(fd)->GetOSFHandle();
+}
+
+int EbrRead(int fd, void* dest, size_t count) {
+    return EbrFile::GetFile(fd)->Read(dest, count);
+}
+
+int EbrWrite(int fd, const void* src, size_t count) {
+    return EbrFile::GetFile(fd)->Write(src, count);
+}
+
+int EbrLseek(int fd, __int64 pos, int whence) {
+    return EbrFile::GetFile(fd)->Lseek(pos, whence);
+}
+
+int EbrTruncate64(int fd, __int64 size) {
+    return EbrFile::GetFile(fd)->Truncate64(size);
+}
+
+size_t EbrTell(int fd) {
+    return EbrFile::GetFile(fd)->Tell();
+}
+
+int EbrFflush(int fd) {
+    return EbrFile::GetFile(fd)->Flush();
+}
+
+bool EbrRemoveEmptyDir(const char* path) {
+    return RemoveDirectoryA(CPathMapper(path));
+}
+
+bool EbrRename(const char* path1, const char* path2) {
+    return rename(CPathMapper(path1), CPathMapper(path2)) == 0;
+}
+
+bool EbrUnlink(const char* path) {
+    return _unlink(CPathMapper(path)) == 0;
+}
+
+#define FSROOT "."
+#define mkdir _mkdir
+char g_WritableFolder[2048] = ".";
+
+void EbrSetWritableFolder(const char* folder) {
+    strcpy_s(g_WritableFolder, folder);
+}
+
+const char* EbrGetWritableFolder() {
+    return g_WritableFolder;
+}
+
+bool EbrGetRootMapping(const char* dirName, char* dirOut, uint32_t maxLen) {
+    if (dirName == NULL) {
+        strcpy_s(dirOut, maxLen, FSROOT);
+        return true;
+    }
+    if (_stricmp(dirName, "Documents") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\Documents", g_WritableFolder);
```

Comment:
>%s\\Documents" [](start = 35, length = 14)

Put these in an internal header?
These are being repeated in other files, thus is would make sense to do so. Also this would make future maintenance easier